grpc/src/builder.rs

```diff
@@ -0,0 +1,189 @@
+use bytes::Bytes;
+use http_body::Body;
+use k256::ecdsa::VerifyingKey;
+use signature::Keypair;
+use tonic::body::BoxBody;
+use tonic::client::GrpcService;
+#[cfg(not(target_arch = "wasm32"))]
+use tonic::transport::{Channel, ClientTlsConfig, Endpoint};
+
+use crate::client::SignerBits;
+use crate::client::StdError;
+use crate::signer::DispatchedDocSigner;
+use crate::DocSigner;
+use crate::GrpcClient;
+use crate::GrpcClientBuilderError;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub struct NativeTransportBits {
+    url: String,
+    load_native_roots: bool,
+    load_webpki_roots: bool,
+}
+
+/// Builder for [`GrpcClient`]
+pub struct GrpcClientBuilder<T> {
+    transport_setup: T,
+    signer_bits: Option<SignerBits>,
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl GrpcClientBuilder<NativeTransportBits> {
+    /// Create a new client connected to the given `url` using [`Channel`] transport
+    ///
+    /// [`Channel`]: tonic::transport::Channel
+    pub fn with_url(url: impl Into<String>) -> Self {
+        GrpcClientBuilder {
+            transport_setup: NativeTransportBits {
+                url: url.into(),
+                load_native_roots: false,
+                load_webpki_roots: false,
+            },
+            signer_bits: None,
+        }
+    }
+
+    /// Enables the platform’s trusted certs.
+    #[cfg(feature = "tls-native-roots")]
+    pub fn with_native_roots(self) -> Self {
+        let NativeTransportBits {
+            url,
+            load_webpki_roots,
+            ..
+        } = self.transport_setup;
+
+        Self {
+            transport_setup: NativeTransportBits {
+                url,
+                load_native_roots: true,
+                load_webpki_roots,
+            },
+            ..self
+        }
+    }
+
+    /// Enables the webpki roots.
+    #[cfg(feature = "tls-webpki-roots")]
+    pub fn with_webpki_roots(self) -> Self {
+        let NativeTransportBits {
+            url,
+            load_native_roots,
+            ..
+        } = self.transport_setup;
+
+        Self {
+            transport_setup: NativeTransportBits {
+                url,
+                load_native_roots,
+                load_webpki_roots: true,
+            },
+            ..self
+        }
+    }
+}
```

Comment:
let's have only `.with_default_tls()` and allow user to use feature flags to control which roots were selected

Reply:
Switched to single function and exact roots being loaded is driven by feature flags 👍